### ISSUE TITLE
[Bug]: assert [0, 1, 2] == [0, 2, 4]

### ISSUE THREAD

**Issue by meiling6226** [vllm-project/vllm-ascend#16519]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Your output of above commands here
```

</details>

### 🐛 Describe the bug

 tests/v1/test_outputs.py
https://github.com/vllm-project/vllm-ascend/actions/runs/34758805062/job/103728096362#annotation:13:15188

=================================== FAILURES 
  ____________________ test_sampling_mask_tensors_from_logits ____________________
  
      def test_sampling_mask_tensors_from_logits():
          tensors = SamplingMaskTensors.from_logits(
              logits=torch.tensor(
                  [
                      [1.0, float("-inf"), 2.0],
                      [3.0, 4.0, float("-inf")],
                      [float("-inf"), 5.0, 6.0],
                  ],
                  device=DEVICE_TYPE,
              ),
              num_sampled_tokens=torch.tensor([1, 0, 1], device=DEVICE_TYPE),
          )
      
          result = tensors.tolists(np.array([1, 0, 1]))
      
          assert result.token_ids.tolist() == [0, 2, 1, 2]
  >       assert result.offsets.tolist() == [0, 2, 4]
  E       assert [0, 1, 2] == [0, 2, 4]
  E         
  E         At index 1 diff: 1 != 2
  E         
  E         Full diff:
  E           [
  E               0,
  E         +     1,
  E               2,
  E         -     4,
  E           ]
  
  tests/v1/test_outputs.py:91: AssertionError
  ___________ test_sampling_mask_matches_processed_top_k_top_p_support ___________
  
      def test_sampling_mask_matches_processed_top_k_top_p_support():
          processed_logits = apply_top_k_top_p(
              logits=torch.tensor(
                  [[6.0, 5.0, 4.0, 4.0, 4.0, 2.0, 1.0, 0.0]], device=DEVICE_TYPE
              ),
              k=torch.tensor([3], device=DEVICE_TYPE),
              p=torch.tensor([0.9], device=DEVICE_TYPE),
          )
          expected_token_ids = (
              torch.isfinite(processed_logits[0]).nonzero().flatten().tolist()
          )
          assert 0 < len(expected_token_ids) < processed_logits.shape[1]
      
          tensors = SamplingMaskTensors.from_logits(
              processed_logits,
              num_sampled_tokens=torch.tensor([1], device=DEVICE_TYPE),
          )
          result = tensors.tolists(np.array([1]))
      
  >       assert result.to_nested_list() == [expected_token_ids]
  E       assert [[0]] == [[0, 1, 3, 4]]
  E         
  E         At index 0 diff: [0] != [0, 1, 3, 4]
  E         
  E         Full diff:
  E           [
  E               [
  E                   0,
  E         -         1,
  E         -         3,
  E         -         4,
  E               ],
  E           ]
  
  tests/v1/test_outputs.py:114: AssertionError

**Comment by jiangyunfan1** [comment]
https://github.com/vllm-project/vllm/pull/54901
